server/src/main/java/org/candlepin/resource/StatusResource.java

```diff
@@ -89,17 +96,23 @@ public StatusResource(RulesCurator rulesCurator,
     @Produces({ MediaType.APPLICATION_JSON})
     @SecurityHole(noAuth = true, anon = true)
     public Status status() {
+        /*
+         * Originally this was used to indicate database connectivity being good/bad.
+         * In reality it could never be false, the request would fail. This check has
+         * been moved to GET /status/db.
+         */
         boolean good = true;
         try {
-            if (rulesCurator != null) {
-                rulesCurator.listAll();
-            }
+            rulesCurator.getUpdatedFromDB();
         }
         catch (Exception e) {
+            log.error("Error checking database connection", e);
             good = false;
         }
+
         Status status = new Status(good, version, release, standalone,
```

Comment:
Current discussions with IT are leaning towards serving a static representation to clients who make this API call, so only running this for IT status checking. For that, we probably need it to be live, though I'm working on getting them to call it less often. Hopefully we can find a way to drop the db query from here entirely, ideally I want something catching exceptions and setting a flag that status will return. If we could do that and just get to this tiny bit of pure java, I'd have to think we're pretty fast. This call as it stands here is 2ms on my machine max.